More beginner-friendly TCP server example (#102)
rust-lang/async-book: src/08_example/04_tests.md

> # Testing Async Code

Reviewer: Sounds pretty general, perhaps we can be more specific that we're testing our code?

# Testing the TCP Server
Let's move on to testing our `handle_connection` function.

> First, we need a `TcpStream` to work with.

Reviewer 2: I'd suggest breaking this paragraph up

```suggestion

First, we need a `TcpStream` to work with.
```

> In an end-to-end or integration test, we might want a to make a real TCP connection

Reviewer 2: ```suggestion
In an end-to-end or integration test, we might want to make a real TCP connection
```

to test our code.
One strategy for doing this could be to connect to `localhost` on port 0.
Port 0 isn't a valid UNIX port, but it'll work for testing.

> The operating system will return a connection on any open TCP port.

Reviewer 2: I don't think the pre-existing text seems quite right here, what about this?

```suggestion
One strategy for doing this is to start a listener on `localhost` port 0.
Port 0 isn't a valid UNIX port, but it'll work for testing.
The operating system will pick an open TCP port for us.
```
